WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -1531,7 +1532,8 @@ private void updateAndSavePostAsync(final AfterSavePostListener listener) {
         }
         mViewModel.updatePostObjectWithUIAsync(mEditPostRepository,
                 this::updateFromEditor,
-                (post) -> {
+                (post, result) -> {
+                    // Ignore the result as we want to invoke the listener even when the PostModel was up-to-date
```

Comment:
I think why we are ignoring the result is not super clear here. That might be because of the name `onPostSave` which doesn't feel like it exactly matches what's happening. Specifically, why we are calling something called `onPostSave` if nothing changed. I am guessing my confusion around this setup may be because the legacy code got refactored, but it was hard to make all parts of it make sense. 

I might be thinking too hard about this though, so I am just expressing my minor confusion in case it helps you to figure out a better way to do this, but please feel free to ignore it completely if it makes sense for someone who is more familiar with what's going on.

Reply:
👍Thanks, good call! I've pushed the fix into `rename-after-post-save-listener`. I'll create a PR in a bit.